### PR TITLE
CORE-4882 Update CI Flow

### PR DESCRIPTION
**Pull request by cthtuf**
### BEFORE MERGING THIS PR

- [x]  We should sync mater and prod branches and merge prod to master. 
- [ ] After merging this PR prod branch will become deprecated and will not be used anymore

New flow, according https://www.notion.so/chainstack/Proposal-Branching-Release-flow-8b60e0d6b6f34c1fb12d4efab51a93ad

Now it's only master and only feature-branches. 
deploy to dev from feature branch
release will be created automatically after merge to master

### CODE REVIEW

**Comment by cthtuf** [comment]
@easeev @akegaviar please take a look, I wanna merge it asap

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,43 +2,76 @@ version: 2
...
-      - test
-
-      - build_dev_image:
+      - check_if_version_updated:
+          context: "dev.chainstack | GCP"
+          filters:
+            branches:
+              only:
+                - /(feature|bug|hotfix|release)\/.*/
+      - check_release_notes_updated:
```
Comment (cthtuf) [inline .circleci/config.yml:13]: It would require to update release notes on each PR

**Comment by easeev** [comment]
@cthtuf @PekopT why 2 people, both not in Engineering, should review the PR with CI changes?

**Comment by cthtuf** [comment]
> @cthtuf @PekopT why 2 people, both not in Engineering, should review the PR with CI changes?

Because you're codeowners here. 
No needs to review CI config, just read description and if you have some concerns - comment please

**Pull request merged**